ansible_ai_connect/ai/api/telemetry/schema1.py

```
@@ -0,0 +1,326 @@
+    request: ContentMatchRequestSerializer | None = None
+    metadata: list = field(factory=list)
+    problem: str = ""
+
+    def set_validated_data(self, validated_data):
+        super().set_validated_data(validated_data)
+        self.request = validated_data
+
+
+# Events associated with the Feedback view
+@define
+class BaseFeedbackEvent(Schema1Event):
+    def set_validated_data(self, validated_data):
+        # This is to deal with a corner case that will be address once
+        # https://github.com/ansible/vscode-ansible/pull/1408 is merged
+        if self.event_name == "inlineSuggestionFeedback" and "inlineSuggestion" in validated_data:
+            event_key = "inlineSuggestion"
+        else:
+            event_key = self.event_name
+        suggestion_quality_data: SuggestionQualityFeedback = validated_data[event_key]
+        super().set_validated_data(suggestion_quality_data)
+
+    @classmethod
+    def init(cls, user, validated_data):
+        mapping = {
+            "inlineSuggestion": schema1.InlineSuggestionFeedbackEvent,
+            "inlineSuggestionFeedback": schema1.InlineSuggestionFeedbackEvent,
+            "suggestionQualityFeedback": schema1.SuggestionQualityFeedbackEvent,
+            "sentimentFeedback": schema1.InlineSuggestionFeedbackEvent,
+            "issueFeedback": schema1.IssueFeedbackEvent,
+            "playbookExplanationFeedback": schema1.PlaybookExplanationFeedbackEvent,
+            "playbookGenerationAction": schema1.PlaybookGenerationActionEvent,
+        }
+        # TODO: handles the key that are at the root level of the structure
+        for key_name, schema1_class in mapping.items():
+            if key_name in validated_data:
+                schema1_event = schema1_class()
+                schema1_event.set_user(user)
+                schema1_event.set_validated_data(validated_data)
+                return schema1_event
+        print("Failed to init BaseFeedbackEvent")
```

Should we not at least use a logger?

I would have thought we should probably raise an exception too (although we hope this line is never reached)?

---

Totally agree, I converted that as a logger.error().